Sources/PackageDescription/PackageDescription.swift

```diff
@@ -640,6 +652,32 @@ func manifestToJSON(_ package: Package) -> String {
 }
 
 var errors: [String] = []
+
+#if os(Windows)
+private var dumpInfo: (package: Package, handle: Int)?
+private func dumpPackageAtExit(_ package: Package, to handle: Int) {
+  let dump: @convention(c) () -> Void = {
+    guard let dumpInfo = dumpInfo else { return }
+
+    let hFile: HANDLE = HANDLE(bitPattern: dumpInfo.handle)!
+    // NOTE: `_open_osfhandle` transfers ownership of the HANDLE to the file
+    // descriptor.  DO NOT invoke `CloseHandle` on `hFile`.
+    let fd: CInt = _open_osfhandle(Int(bitPattern: hFile), _O_APPEND)
+    // NOTE: `_fdopen` transfers ownership of the file descriptor to the
+    // `FILE *`.  DO NOT invoke `_close` on the `fd`.
+    guard let fp = _fdopen(fd, "w") else {
+      _close(fd)
+      return
+    }
+    defer { fclose(fp) }
```

Comment:
closes fd as well?

Reply:
Precisely.  That's the note on L666-667.  `_fdopen` will take over the `fd` so we should not call `_close` on the fd.  If there is a way to make that clearer (and safer) I'm open to improving this further.  That dance is very subtle.  I've tried to document it in comments and the commit message, but if there is a better way to prevent confusion, I'm happy to work on that.